app/services/recurring_contribution/update_status.rb

```diff
@@ -13,6 +14,8 @@ def self.process(request, resource)
 
   def process
     update_status if request_source_is_pagarme?
+  rescue ArgumentError
```

Comment:
se o status for passado invalido o model deve retornar algum erro de validação e nao um `ArgumentError`, nesse caso vc quer um record invalid